### PR TITLE
Split possible root_span into separate aggregated traces

### PR DESCRIPTION
**Pull request by alanpo1**
This PR is focused on adding logic to separate a single trace_id group from a "forest" of spans into individual trees in situations where we have multiple root_span in a single trace_id grouping.

local e2e using Vivienne's dataset: https://ml.azure.com/runs/208f275c-a85e-4399-a8d9-0f57fba60abf?wsid=/subscriptions/1aefdc5e-3a7c-4d71-a9f9-f5d3b03be19a/resourcegroups/yuachengtestrg/workspaces/ai-proj-eastus&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### CODE REVIEW

**Review by RichardLi1437 — APPROVED** [review]

File: assets/model_monitoring/components/src/shared_utilities/span_tree_utils.py
Comment (alanpo1) [reply assets/model_monitoring/components/src/shared_utilities/span_tree_utils.py:204]: I suppose the only advantage is for keeping the root_span property logic simple but I will look into changing the code as you mentioned in a follow up PR since some of Vivienne's e2e testing is blocked until this PR is merged.

**Pull request merged**
